app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.length = 0
+        self.hash_table: list[None | list[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
```

Comment:
Instead of using 'any' for type hinting, consider using 'Any' from the typing module for better clarity and consistency.